spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'region': bindings['TEST_GCE_REGION'],
+        'regions': [bindings['TEST_GCE_REGION']],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'user': '[anonymous]'
+      }
+    ]
+
+
+  def _make_bs_link(self, bs):
+    '''Make a fully-formatted backend service link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/backendServices/' + bs)
+
+
+  def _make_hc_link(self, hc):
+    '''Make a fully-formatted health check link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/httpHealthChecks/' + hc)
+
+
+  def upsert_full_load_balancer(self):
+    '''Upserts L7 LB with full hostRules, pathMatchers, etc.
+
+    Calls the upsertLoadBalancer operation with a payload, then verifies that
+    the expected resources are visible on GCP.
+    '''
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=self.__default_upsert,
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    full_hc_name = self.__default_hc['name']
+    hc_link = self._make_hc_link(self.__default_hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (contract_builder.new_clause_builder('Health Checks Created',
+                                         retryable_for_secs=30)
+     .list_resource('httpHealthChecks')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', full_hc_name),
+         jp.PathContainsPredicate('requestPath', '/'),
+         jp.PathContainsPredicate('port', 80)
+       ]
+     )
+    )
+    (contract_builder.new_clause_builder('Backend Services Created',
+                                         retryable_for_secs=30)
+     .list_resource('backendServices')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', 'backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     ).contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', 'default-backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     ).contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', 'pm-backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     )
+    )
+    (contract_builder.new_clause_builder('Url Map Created',
+                                        retryable_for_secs=30)
+     .list_resource('urlMaps')
+     .contains_pred_list(
```

Comment:
just to be clear, you cannot say that now. I'm speculating on whether I should enable you to.